src/payment/strategies/barclaycard/barclaycard-payment-strategy.ts

```diff
@@ -0,0 +1,54 @@
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { OrderActionCreator,  OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import { PaymentArgumentInvalidError } from '../../errors';
+import PaymentActionCreator from '../../payment-action-creator';
+import { PaymentRequestOptions } from '../../payment-request-options';
+import * as paymentStatusTypes from '../../payment-status-types';
+import PaymentStrategy from '../payment-strategy';
+
+export default class BarclaycardPaymentStrategy implements PaymentStrategy {
+    constructor(
+        private _store: CheckoutStore,
+        private _orderActionCreator: OrderActionCreator,
+        private _paymentActionCreator: PaymentActionCreator
+    ) { }
+
+    async execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment } = payload;
+        const paymentData = payment && payment.paymentData;
+
+        if (!payment) {
+            throw new PaymentArgumentInvalidError(['payment']);
+        }
+
+        await this._store.dispatch(this._orderActionCreator.submitOrder(payload, options));
+
+        return await this._store.dispatch(this._paymentActionCreator.initializeOffsitePayment(
+            payment.methodId,
+            payment.gatewayId,
+            {
+                ...paymentData,
```

Comment:
Just out of curiosity, why are we destructuring this?

Reply:
@icatalina  Im working on a very similar PR (#746 ), and it's more likely for that PR to be merged first, so I was thinking into making them as similar as possible to have fewer conflicts to solve when the other PR gets merged.

Reply:
Im working on a PR that modifies the same files (#746) that is more likely to be merged  before this one, so  I'm trying to keep files similar to have the fewer conflicts as possible